.github/workflows/vertexai.yml

```diff
@@ -15,20 +15,42 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  spm-package-resolved:
+    runs-on: macos-14
+    outputs:
+      package_resolved_hash: ${{ steps.cache.outputs.cache-primary-key }}
+    env:
+      FIREBASECI_USE_LATEST_GOOGLEAPPMEASUREMENT: 1
+    steps:
+      - uses: actions/checkout@v4
+      - name: Generate Swift Package.resolved
+        id: swift_package_resolve
+        run: |
```

Comment:
Do we need to test if there is a cache hit - like in the `Install Dependencies` at https://github.com/actions/cache/blob/main/caching-strategies.md#step-2---restore-the-built-artifact-from-cache-using-the-same-key-and-path?

Reply:
Potentially though the cache hit test would be in the jobs below (after attempting to restore).

In the caching phase here we could probably run:
```
- name: Xcode
      run: sudo xcode-select -s /Applications/${{ matrix.xcode }}.app/Contents/Developer
    - name: Initialize xcodebuild
      run: scripts/setup_spm_tests.sh
```

Will need some refactoring to do that.

Reply:
As long as we're regenerating the Package.resolved, I'm not sure it matters.